spacy/ml/models/entity_linker.py

```diff
@@ -77,24 +83,46 @@ def span_maker_forward(model, docs: List[Doc], is_train) -> Tuple[Ragged, Callab
     return outputs, lambda x: []
 
 
+@registry.misc("spacy.KBType.v1")
+def get_kb_type() -> Type[BaseKnowledgeBase]:
+    """Returns KB type."""
+    return KnowledgeBase
```

Comment:
This is IMO useful because even though we have the KB loader in the `initialize` section, the default loader function `load_kb()` doesn't know which type of `BaseKnowledgeBase` to initialize. 
So either we require the user to provide custom implementations of KB loader and `empty_kb`, or we make them specify the KB type and automatically load/initialize the correct KB type.

Since the loading and empty initialization is boilerplate as long as no additional arguments are needed, this seems more intuitive to me. We could also drop this and require custom loaders and initializers for custom KBs though.

Reply:
Hmmm this looks odd to me, but perhaps I'm not fully understanding the issue. Ideally each KB should be able to load/initialize itself, no? The `load_kb` option is currently tied to our specific default Cython implementation, and that should probably stay as such. Agreed it's pretty generic but it's also very little effort to add if you're already reimplementing a KB?

Reply:
I agree this is pretty weird looking. If the only intended use of this is related to initialization (which isn't obvious to me from the registry declaration / function here) then I think it makes sense to just register the initialization stuff directly, even if it's a little boilerplate.

Reply:
Fair enough, it is very little effort to write those. Maybe saving that little amount of work doesn't justify introducing a new config option. Will revert.